services/uibackend/main.go

```diff
@@ -0,0 +1,82 @@
+package main
+
+import (
+	"crypto"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/sha256"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/json"
+	"encoding/pem"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/gorilla/mux"
+)
+
+type Onboarding struct {
+	Token string `json:"token"`
+}
+
+func parseKey(s []byte) *rsa.PrivateKey {
+
+	Block, _ := pem.Decode(s)
+	k, err := x509.ParsePKCS1PrivateKey(Block.Bytes)
+	if err != nil {
+		panic(err)
```

Comment:
Shouldn't we log an error here? also why are you panicing the entire server? shouldn't we just have an HTTP reply with an Interal error code?